web/src/compose.js

```diff
@@ -358,44 +359,71 @@ export function render_and_show_preview($preview_spinner, $preview_content_box,
         } else {
             rendered_preview_html = rendered_content;
         }
-
-        $preview_content_box.html(util.clean_user_content_links(rendered_preview_html));
+        if (message_number) {
+            // a valid non-zero number means that the message should be enumerated
+            const params = {
+                message_number,
+                rendered_preview_html,
+            };
+            const enumerated_rendered_preview_html = render_enumerated_split_message_part(params);
+            $preview_content_box.append(
+                $(util.clean_user_content_links(enumerated_rendered_preview_html)),
+            );
+        } else {
+            $preview_content_box.html(util.clean_user_content_links(rendered_preview_html));
+        }
         rendered_markdown.update_elements($preview_content_box);
     }
 
     if (content.length === 0) {
         show_preview($t_html({defaultMessage: "Nothing to preview"}));
     } else {
-        if (markdown.contains_backend_only_syntax(content)) {
-            const $spinner = $preview_spinner.expectOne();
-            loading.make_indicator($spinner);
-        } else {
-            // For messages that don't appear to contain syntax that
-            // is only supported by our backend Markdown processor, we
-            // render using the frontend Markdown processor (but still
-            // render server-side to ensure the preview is accurate;
-            // if the `markdown.contains_backend_only_syntax` logic is
-            // wrong, users will see a brief flicker of the locally
-            // echoed frontend rendering before receiving the
-            // authoritative backend rendering from the server).
-            markdown.render(content);
+        render_message_part(content, 1);
+        function render_message_part(whole_content, message_number) {
+            const [content, remaining_content] = compose_validate.split_message(whole_content);
+            if (!remaining_content && message_number === 1) {
+                // if no content remains on splitting the message once,
+                // it means that the message will not be split into parts
+                // and we should not enumerate the parts.
+                message_number = Number.NaN;
```

Comment:
Consider using 0 instead. It might be simpler while working just as well

Reply:
I was earlier using 0, but replaced it with NaN for lexical clarity.
I'll replace it again with 0.